src/Datastore/RAMDataStore.cpp

```diff
@@ -151,11 +151,11 @@ TResult<bool> RAMDataStore::hasTrack(TTrackID ID, QueueType q) {
   }
 }
 
-TResultOpt RAMDataStore::voteTrack(TSessionID sID, TTrackID tID, TVote vote) {
+TResultOpt RAMDataStore::voteTrack(TSessionID const &sID, TTrackID const &tID, TVote vote) {
   // Exclusive Access to Song Queue and User
   unique_lock<shared_mutex> MyLockQueue(mQueueMutex, defer_lock);
   unique_lock<shared_mutex> MyLockUser(mUserMutex, defer_lock);
-  scoped_lock(MyLockQueue, MyLockUser);
+  scoped_lock Lock(MyLockQueue, MyLockUser);
```

Comment:
`std::lock`
https://en.cppreference.com/w/cpp/thread/lock

Reply:
why would i need the regular one here? now that i actually keep the created object that should work, right?

Reply:
`std::lock` and `std::scoped_lock` do not do the same at all (at least conceptionally).

`scoped_lock` is a CLASS which takes mutexes to lock when the object is created and unlocks them when it goes out of scope
`lock` is a FUNCTION which takes lockable objects (not mutexes per se!) and locks them while avoiding deadlocks

in the end both approaches do the same but are conceptionally very different (if you look through the examples on cppreference closly you notice that)

Reply:
besides, your approach only works because `unique_lock` happens to have all the methods which are needed by `scoped_lock` to work (which may or may not be intended)